cleverhans/attacks_tf.py

```diff
@@ -417,3 +417,245 @@ def jacobian_augmentation(sess, x, X_sub_prev, Y_sub, grads, lmbda,
 
     # Return augmented training data (needs to be labeled afterwards)
     return X_sub
+
+
+class CarliniWagnerL2:
```

Comment:
What's the purpose of having two different classes?

Also, if we keep the two-class version, I suspect this one should inherit from `object` to make it a "new-style class".